### ISSUE TITLE
ReturnType of `CommitBaseWithFiles` incorrect

### ISSUE THREAD

**Issue by ninesunsabiu** [domharrington/node-gitlog#82]
https://github.com/domharrington/node-gitlog/blob/617373ef33158760cd4b5d3d4b4a4e3b887448fe/src/index.ts#L312-L315

the type of `CommitBaseWithFiles['status'] is string`, but it is `Array<string>` in fact.

https://github.com/domharrington/node-gitlog/blob/617373ef33158760cd4b5d3d4b4a4e3b887448fe/src/index.ts#L209-L214

The code here proves this as well

**Comment by cmj2002** [comment]
I also encountered this issue and had to use `@ts-ignore` to ignore it. @domharrington Why did you close the PR?

**Comment by domharrington** [comment]
I was doing some cleanup and maybe I removed it unintentionally. Would you mind sending me a code sample to demonstrate the error and I'll reapply the patch if it fixes it? Thanks in advance.

**Comment by cmj2002** [comment]
> I was doing some cleanup and maybe I removed it unintentionally. Would you mind sending me a code sample to demonstrate the error and I'll reapply the patch if it fixes it? Thanks in advance.

A reproduction is in https://github.com/cmj2002/reproduce-gitlog-82

`npm start` outputs:

```
[Commits] [
  {
    status: [ 'M' ],
    files: [ 'README.md' ],
    authorDate: '2024-07-15 14:27:21 +0000',
    authorEmail: 'me@caomingjun.com'
  },
  {
    status: [ 'A' ],
    files: [ 'README.md' ],
    authorDate: '2024-07-15 21:03:28 +0800',
    authorEmail: 'me@caomingjun.com'
  }
]
[Authors] [ 'me@caomingjun.com' ]
[Created] 2024-07-15 21:03:28 +0800
[Modified] 2024-07-15 14:27:21 +0000
```

If remove [line 18 in `src/main.ts`](https://github.com/cmj2002/reproduce-gitlog-82/blob/d594816c93599807f2472074593dfb81a9c5d347/src/main.ts#L18), it will throw a error when compiling:

```
src/main.ts:19:7 - error TS2322: Type 'string[]' is not assignable to type 'string'.

19       status: ["A"],
         ~~~~~~


Found 1 error in src/main.ts:19
```

**Comment by domharrington** [comment]
This has been fixed in v5.0.2! https://www.npmjs.com/package/gitlog/v/5.0.2 Thanks for the reproduction steps. Let me know if you have any other problems!
